out/out.go

```diff
@@ -174,12 +174,23 @@ func startSchemaCreationOrUpdate(appsyncClient *appsync.AppSync, schemaCreatePar
 	err := req.Send()
 	if err != nil {
 		return err
-
 	}
 	status := *resp.Status
-	if status == "PROCESSING" {
-		time.Sleep(time.Second * 3)
+
+	for status == "PROCESSING" {
+		time.Sleep(3 * time.Second)
+
+		creationStatusOutput, err := appsyncClient.GetSchemaCreationStatus(&appsync.GetSchemaCreationStatusInput{
+			ApiId: schemaCreateParams.ApiId,
```

Comment:
Can you use the private `getSchemaCreationStatus` functions which called `appsyncClient.GetSchemaCreationStatus` 
the reason is because that function log the error also will have one-way int he code to check the creation schema status.
However, the need to modify the `getSchemaCreationStatus` to return an error maybe like this:
```
func getSchemaCreationStatus(appsyncClient *appsync.AppSync, schemaStatusParams *appsync.GetSchemaCreationStatusInput, logger *log.Logger) (string, string, error) {
	StatusOutput, err := appsyncClient.GetSchemaCreationStatus(schemaStatusParams)
	if err != nil {
		logger.Println("Failed to get Schema Creation status, However the Schema creation might be succeeded, check the AWS console and re-tigger the build if the schema not created/updated: %s", err)
		return "", "", err
	}
	creationStatus := *StatusOutput.Status
	creationDetails := *StatusOutput.Details

	return creationStatus, creationDetails, nil
}
```